workspace/src/common/launch/launch_utils/launch_utils/utilities.py

```diff
@@ -0,0 +1,172 @@
+    but this returns the _install_ directory where everything is put after colcon build. In some
+    cases, we want to put code/generated data/etc. in the _source_ code. This package helps with this.
+
+    It takes a package name (str) and a base name (str). The base name is the location to begin
+    an downward search of the file tree for the package name. If base name is None (the default),
+    the environment variable 'ROS_WORKSPACE_SRC' will be used. If 'ROS_WORKSPACE_SRC' is not found,
+    the current working directory is used.
+
+    It is considered found if the package name is in the current searched folder absolute path
+    and a package.xml file is found in the same folder.
+
+    Will return the path as a pathlib.Path object or None if not found.
+
+    WARNING: Not fast if there are a lot of folders on the system.
+    """
+    import os
+    from pathlib import Path
+
+    if base is None:
+        base = os.environ.get("ROS_WORKSPACE_SRC", os.getcwd())
+
+    for root, dirs, files in os.walk(base):
+        if package == os.path.basename(root) and 'package.xml' in files:
+            return Path(root)
+
+    return None
+
+
+def AddComposableNode(
+    ld: LaunchDescription,
+    *,
+    plugin: str,
+    executable: str,
+    composable_conditions: "List[List[SomeSubstitutionType]" = [],
+    node_conditions: "List[List[SomeSubstitutionType]]" = [],
+    container: str = "container",
+    composable_node_kwargs: dict = {},
+    load_composable_nodes_kwargs: dict = {},
+    node_kwargs: dict = {},
+    add_as_regular_node: bool = True,
+    **kwargs
+) -> None:
+    """
+    Conditionally add a node dependent on whether the LaunchConfiguration ``container`` is
+    not "". If not "", it will load the node into the composable container using ``LoadComposableNodes``.
+    In this case, "use_intra_process_comms" will be set to true. If ``container`` is "",
+    it will simply load the node using ``Node`` without intraprocess comms.
+
+    If ``add_as_regular_node`` is set to False (defaults to True), it will not create a ``Node`` if ``container`` is unset.
+    """
+
+    from launch_utils.conditions import MultipleIfConditions
+    from launch_utils.substitutions import QuoteWrappedPythonExpression
+
+    from launch_ros.actions import Node, LoadComposableNodes
+    from launch_ros.descriptions import ComposableNode
+
+    container = GetLaunchArgument(container)
+
+    # Add a node the container only if the container launch argument is set
+    composable_condition = QuoteWrappedPythonExpression([container, " != ''"])
+    composable_nodes = [
+        ComposableNode(
+            plugin=plugin,
+            extra_arguments=[{"use_intra_process_comms": True}],
+            **composable_node_kwargs,
+            **kwargs,
+        )
+    ]
+    load_composable_nodes = LoadComposableNodes(
+        condition=MultipleIfConditions(
+            [composable_condition, *composable_conditions]),
```

Comment:
@StefanCaldararu `MultipleIfConditions` is used here. I wrote it to support unknown conditions at runtime. Like there could be a lot of conditions, and so writing an inline string generator with the proper conditions seems a little much. Do you think it would make sense to delete it?

Reply:
Ok sounds good, I'm not sure I entirely understand but I'm sure it'll make more sense once there is documentation...